plugins/Actions/VisitorDetails.php

```diff
@@ -160,6 +161,14 @@ public function extendActionDetails(&$action, $nextAction, $visitorDetails)
             unset($action['url_prefix']);
         }
 
+        if (array_key_exists('url', $action) && strpos($action['url'], 'http://') === 0) {
+            $host = parse_url($action['url'], PHP_URL_HOST);
+
+            if ($host && $this->shouldUseHttpsHost($visitorDetails['idSite'], $host)) {
+                $action['url'] = 'https://' . substr($action['url'], 7);
```

Comment:
Might be good for better readability to not hard code `7` but instead use like `strlen('http://')`.

Also we might want to use Common::mb_substr here?

Reply:
Will add a comment, so it's clear